### ISSUE TITLE
[ONNX] Lossing Dropout layer when export pytorch model to onnx 

### ISSUE THREAD

**Issue by xiaoyuan0203** [pytorch/pytorch#80947]
### 🐛 Describe the bug

I try to export Alexnet model to onnx

```python
import torch
from torchvision.models.alexnet import *

dummy_input = torch.randn(10, 3, 224, 224)
alexnet_model = alexnet(weights=AlexNet_Weights.IMAGENET1K_V1)

torch.onnx.export(alexnet_model, dummy_input, "alexnet.onnx",
                  input_names=['input'],
                  output_names=['output'],  # verbose=True,
                  )
```

Then view the exported onnx
```python
import onnx

# Load the ONNX model
model = onnx.load("alexnet.onnx")

# Check that the model is well formed
onnx.checker.check_model(model)

# Print a human readable representation of the graph
print(onnx.helper.printable_graph(model.graph))
```

But get:

```
...............................................................................(skip)
>>>  %onnx::Flatten_30 = AveragePool[kernel_shape = [1, 1], strides = [1, 1]](%input.28)
>>>  %input.32 = Flatten[axis = 1](%onnx::Flatten_30)
>>>  %input.36 = Gemm[alpha = 1, beta = 1, transB = 1](%input.32, %classifier.1.weight, %classifier.1.bias)
>>>  %onnx::Gemm_33 = Relu(%input.36)
>>>  %input.40 = Gemm[alpha = 1, beta = 1, transB = 1](%onnx::Gemm_33, %classifier.4.weight, %classifier.4.bias)
>>>  %onnx::Gemm_35 = Relu(%input.40)
>>>  %output = Gemm[alpha = 1, beta = 1, transB = 1](%onnx::Gemm_35, %classifier.6.weight, %classifier.6.bias)
>>>  return %output
}
```

File alexnet.onnx viewed by Netron is following:

![alexnet](https://user-images.githubusercontent.com/78028879/177519352-112ccc8c-98c9-415c-9353-85bf6a250a69.png)

where Gemm means Generally Matrix Multiplication is a FC layer

The exported onnx lacks Dropout layer, while Alexnet looks like:

```
AlexNet(
  (features): Sequential(
    (0): Conv2d(3, 64, kernel_size=(11, 11), stride=(4, 4), padding=(2, 2))
    (1): ReLU(inplace=True)
    (2): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
    (3): Conv2d(64, 192, kernel_size=(5, 5), stride=(1, 1), padding=(2, 2))
    (4): ReLU(inplace=True)
    (5): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
    (6): Conv2d(192, 384, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    (7): ReLU(inplace=True)
    (8): Conv2d(384, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    (9): ReLU(inplace=True)
    (10): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    (11): ReLU(inplace=True)
    (12): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
  )
  (avgpool): AdaptiveAvgPool2d(output_size=(6, 6))
  (classifier): Sequential(
    (0): Dropout(p=0.5, inplace=False)
    (1): Linear(in_features=9216, out_features=4096, bias=True)
    (2): ReLU(inplace=True)
    (3): Dropout(p=0.5, inplace=False)
    (4): Linear(in_features=4096, out_features=4096, bias=True)
    (5): ReLU(inplace=True)
    (6): Linear(in_features=4096, out_features=1000, bias=True)
  )
)
```

### Versions

PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 家庭中文版
GCC version: (GCC) 11.2.0
Clang version: Could not collect
CMake version: version 3.22.3
Libc version: N/A

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:59:45) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22598-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.23.0
[pip3] pytorch-lightning==1.6.4
[pip3] segmentation-models-pytorch==0.2.1
[pip3] torch==1.12.0
[pip3] torchaudio==0.12.0
[pip3] torchkeras==2.1.2
[pip3] torchmetrics==0.9.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.13.0
[conda] blas                      2.115                       mkl    conda-forge
[conda] blas-devel                3.9.0              15_win64_mkl    conda-forge
[conda] cpuonly                   2.0                           0    pytorch
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] libblas                   3.9.0              15_win64_mkl    conda-forge
[conda] libcblas                  3.9.0              15_win64_mkl    conda-forge
[conda] liblapack                 3.9.0              15_win64_mkl    conda-forge
[conda] liblapacke                3.9.0              15_win64_mkl    conda-forge
[conda] mkl                       2022.1.0           h6a75c08_874    conda-forge
[conda] mkl-devel                 2022.1.0           h57928b3_875    conda-forge
[conda] mkl-include               2022.1.0           h6a75c08_874    conda-forge
[conda] numpy                     1.20.3                   pypi_0    pypi
[conda] pytorch                   1.12.0              py3.8_cpu_0    pytorch
[conda] pytorch-lightning         1.6.4                    pypi_0    pypi
[conda] pytorch-mutex             1.0                         cpu    pytorch
[conda] segmentation-models-pytorch 0.2.1                    pypi_0    pypi
[conda] torchaudio                0.12.0                 py38_cpu    pytorch
[conda] torchkeras                2.1.2                    pypi_0    pypi
[conda] torchmetrics              0.9.1                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.13.0                 py38_cpu    pytorch

进程已结束,退出代码0

**Comment by xiaoyuan0203** [comment]
It seems that when `training` parameter for `torch.onnx.export` function is `TrainingMode.EVAL`(Default), Dropout Layer will be ignored. Of course it is.
